### ISSUE TITLE
@As drops last character in particular case

### ISSUE THREAD

**Issue by nikowitt** [TNG/JGiven#118]
Hi Jan,

some small thing I've just observed:
```
	@As("$ entity [with restrictions $]")
	public T entity_of_$_with_restrictions_$(@SimpleClassName Class<? extends Base<?>> clazz,
			Criterion... restrictions)
			throws Exception {
(...)
	}
```
which is called via
```
.entity_of_$_with_restrictions_$(MetaDocument.class, Restrictions.isEmpty(MetaDocument.TRANSMITTALS),
						Restrictions.gt(Base.NUMBER_OF_ATTACHMENTS, 0))
```

results in the report
```
     And metadocument entity [with restrictions transmittals is empty, numberOfAttachments>0
```
The last "]" is dropped. When I'm using

```
	@As("$ entity [with restrictions $ ]")
```
with a blank after $, the report is printed out correctly:

```
     And metadocument entity [with restrictions transmittals is empty, numberOfAttachments>0 ]
```

Best regards,
Niko

**Comment by janschaefer** [comment]
Ok. This is actually a 'feature' of JGiven. This allows you to give the placeholders names. So when you can write @As("$entity entity") and the whole word `$entity` will be replaced by the placeholder. In this case, JGiven treats `$]` as the placeholder to replace.

**Comment by janschaefer** [comment]
I think I could change the behavior here to only allow [a-zA-Z0-9] as a valid character for a placeholder name.

**Comment by janschaefer** [comment]
I think it is also fair to treat this as a bug ;-)

**Comment by nikowitt** [comment]
Either that or a minimum length of at least 2 characters are required, but maybe it makes sense to define a single char name like $a, so your suggestion is probably the best solution.

Am 11. September 2015 17:06:08 MESZ, schrieb Jan <notifications@github.com>:

**Comment by janschaefer** [comment]
I actually decided to rewrite most of the placeholder parsing, which introduces a small backwards incompatibility. Thus this fix will be released with 0.9.0.
